src/toil/fileStores/cachingFileStore.py

```diff
@@ -1702,7 +1702,7 @@ def startCommit(self, jobState=False):
         self.waitForCommit()
 
         # Start the commit thread
-        self.commitThread = threading.Thread(target=self.startCommitThread, args=(jobState,))
+        self.commitThread = threading.Thread(target=self.startCommitThread, args=(jobState,), daemon=True)
```

Comment:
The commit thread runs `startCommitThread()` which just runs through and doesn't loop waiting to talk to other threads. So I think this shouldn't actually be a daemon.